internal/executor/executor.go

```diff
@@ -139,9 +141,7 @@ func (executor *Executor) RunBuild() {
 		}
 		backgroundCommand.Logs.Finalize()
 	}
-	if executor.commandTo == "" {
-		// this was a separate container or the last stage of a pipe
-		// we can clean up working directory now
+	if executor.cleanWorkingDir {
```

Comment:
Shouldn't we still check for `executor.commandTo ==""` (see comment above)?

Reply:
Actually just removed the comment. It's not accurate. In a dirty mode `commandTo` will also be unset

Reply:
I think the `-clean-working-dir` will only be used by persistent worker mode with no isolation. Otherwise environment is ephemeral  and there is no need to clean anything, the whole VM or container will be cleaned up.